.github/workflows/CI.yml

```
@@ -230,6 +230,10 @@ jobs:
       run: (cd tracing-core && cargo test --no-default-features)
     - name: "Test tracing no-std support"
       run: (cd tracing && cargo test --no-default-features)
+      # this skips running doctests under the `--no-default-features` flag,
+      # as rustdoc isn't aware of cargo's feature flags.
+    - name: "Test tracing-subscriber with all features disabled"
+      run: (cd tracing-subscriber && cargo test --lib --tests --no-default-features)
```

note: i _believe_  that if we switched the workspace to use [`resolver = "2"`](https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions), we wouldn't have to `cd` into each crate's directory. but, this only works on Rust 1.51+, so we would have to wait until our MSRV is 1.51 (e.g., we can't make that change today).

it _might_ be worth leaving a TODO comment for that, though...

---

oh, another thought. while we're changing this, we may want to change this to use the github actions `working-directory` key, rather than `(cd tracing-subscriber && ...)`. this is what the `cargo hack` checks already do:
https://github.com/tokio-rs/tracing/blob/005b97a21c94c4dec555cf6cd6d0a3068e3325fc/.github/workflows/CI.yml#L72
```suggestion
      run: cargo test --lib --tests --no-default-features
      working-directory: tracing-subscriber
```
We may want to change the other runs in this CI job as well. However, we can also do this in a separate branch.

---

Added a todo assigned to myself!